### PR TITLE
[SYCLomatic] Migrate CU_JIT* options

### PR DESCRIPTION
**Pull request by LU-JOHN**
Migrate CU_JIT* options that are passed to cuLoadDataEx.

Signed-off-by: Lu, John [john.lu@intel.com](mailto:john.lu@intel.com)

### CODE REVIEW

File: clang/lib/DPCT/ASTTraversal.cpp [hunk 1]
```diff
@@ -4052,6 +4052,31 @@ void FFTEnumsRule::runRule(const MatchFinder::MatchResult &Result) {
...
+          to(enumConstantDecl(matchesName(
+              "(CU_JIT_*)"))))
+          .bind("CU_JITConstants"),
+      this);
+}
+
+void CU_JITEnumsRule::runRule(const MatchFinder::MatchResult &Result) {
+  if (const DeclRefExpr *DE =
+          getNodeAsType<DeclRefExpr>(Result, "CU_JITConstants")) {
+    auto *EC = cast<EnumConstantDecl>(DE->getDecl());
```
Comment (0x12CC) [inline clang/lib/DPCT/ASTTraversal.cpp]: Can this declaration be removed?

File: clang/lib/DPCT/ASTTraversal.cpp [hunk 2]
```diff
@@ -4052,6 +4052,31 @@ void FFTEnumsRule::runRule(const MatchFinder::MatchResult &Result) {
...
+    emplaceTransformation(new ReplaceStmt(DE, "0"));
+
+    report(DE->getBeginLoc(),
+           Diagnostics::HOSTALLOCMACRO_NO_MEANING,
+           true, DE->getDecl()->getNameAsString());
+    return;
```
Comment (0x12CC) [inline clang/lib/DPCT/ASTTraversal.cpp]:
```suggestion
```

File: clang/lib/DPCT/ASTTraversal.cpp [hunk 3]
```diff
@@ -2167,7 +2167,7 @@ void TypeInDeclRule::registerMatcher(MatchFinder &MF) {
                   "cufftReal", "cufftDoubleReal", "cufftComplex",
                   "cufftDoubleComplex", "cufftResult_t", "cufftResult",
                   "cufftType_t", "cufftType", "thrust::pair", "CUdeviceptr",
-                  "cudaDeviceAttr", "CUmodule", "CUfunction", "cudaMemcpyKind",
+                  "cudaDeviceAttr", "CUmodule", "CUjit_option", "CUfunction", "cudaMemcpyKind",
```
Comment (0x12CC) [inline clang/lib/DPCT/ASTTraversal.cpp]: Use `clang-format` since this line now exceeds 80 columns.

File: clang/test/dpct/cu_jit.cu [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+//CHECK:a[25] = 0;
+  a[25] = CU_JIT_INPUT_LIBRARY;
+
+//CHECK:DPCT1048:{{[0-9]+}}: The original value CU_JIT_NUM_INPUT_TYPES is not meaningful in the migrated code and was removed or replaced with 0. You may need to check the migrated code.
+//CHECK:a[26] = 0;
+  a[26] = CU_JIT_NUM_INPUT_TYPES;
+
+//CHECK:a[27] = CU_JIT_NOT_A_CUDA_OPTION;
+  a[27] = CU_JIT_NOT_A_CUDA_OPTION;
+}
```
Comment (0x12CC) [inline clang/test/dpct/cu_jit.cu]:
```suggestion
}

```

**Review by 0x12CC — CHANGES_REQUESTED** [review]

File: clang/lib/DPCT/ASTTraversal.cpp (hunk 1 above)
Comment (LU-JOHN) [reply clang/lib/DPCT/ASTTraversal.cpp]: EC removed

File: clang/lib/DPCT/ASTTraversal.cpp (hunk 2 above)
Comment (LU-JOHN) [reply clang/lib/DPCT/ASTTraversal.cpp]: return unnecessary and removed

File: clang/lib/DPCT/ASTTraversal.cpp (hunk 3 above)
Comment (LU-JOHN) [reply clang/lib/DPCT/ASTTraversal.cpp]: reformatted

File: clang/test/dpct/cu_jit.cu (hunk 4 above)
Comment (LU-JOHN) [reply clang/test/dpct/cu_jit.cu]: done

**Review by 0x12CC — APPROVED** [review]

**Review by jzc — APPROVED** [review]

**Pull request merged**